tests/OAuth2SummitTicketTypesApiTest.php

```diff
@@ -154,6 +162,176 @@ public function testGetAllowedTicketTypes(){
+            'per_page' => 10,
+            'order'    => '+id',
+            'filter'   => 'promo_code==' . self::$default_discount_code->getCode(),
+        ];
+
+        $headers = [
+            "HTTP_Authorization" => " Bearer " . $this->access_token,
+            "CONTENT_TYPE"        => "application/json"
+        ];
+
+        $response = $this->action(
+            "GET",
+            "OAuth2SummitsTicketTypesApiController@getAllowedBySummitAndCurrentMember",
+            $params,
+            [],
+            [],
+            [],
+            $headers
+        );
+
+        $content = $response->getContent();
+        $this->assertResponseStatus(200);
+        $ticket_types = json_decode($content);
+        $this->assertTrue(!is_null($ticket_types));
+        $this->assertNotEmpty($ticket_types->data);
+        $this->assertLessThan($ticket_types->data[0]->cost, $ticket_types->data[0]->cost_with_applied_discount);
+        $this->assertEquals(SummitTicketType::Subtype_Regular, $ticket_types->data[0]->sub_type);
+        return $ticket_types;
+    }
+
+    public function testGetInvitationTicketTypesApplyingDiscountCode(){
+        $invitation = new SummitRegistrationInvitation();
+        $invitation->setFirstName(self::$member->getFirstName());
+        $invitation->setLastName(self::$member->getLastName());
+        $invitation->setEmail(self::$member->getEmail());
+        $invitation->addTicketType(self::$default_ticket_type_3);
+        self::$summit->addRegistrationInvitation($invitation);
+
+        self::$default_discount_code->addAllowedTicketType(self::$default_ticket_type_3);
+        self::$em->persist(self::$default_discount_code);
+        self::$em->persist(self::$summit);
+        self::$em->flush();
+
+        $params = [
+            'id'       => self::$summit->getId(),
+            'page'     => 1,
+            'per_page' => 10,
+            'order'    => '+id',
+            'filter'   => 'promo_code==' . self::$default_discount_code->getCode(),
+        ];
+
+        $headers = [
+            "HTTP_Authorization" => " Bearer " . $this->access_token,
+            "CONTENT_TYPE"        => "application/json"
+        ];
+
+        $response = $this->action(
+            "GET",
+            "OAuth2SummitsTicketTypesApiController@getAllowedBySummitAndCurrentMember",
+            $params,
+            [],
+            [],
+            [],
+            $headers
+        );
+
+        $content = $response->getContent();
+        $this->assertResponseStatus(200);
+        $ticket_types = json_decode($content);
+        $this->assertTrue(!is_null($ticket_types));
+        $this->assertNotEmpty($ticket_types->data);
+        $this->assertLessThan($ticket_types->data[0]->cost, $ticket_types->data[0]->cost_with_applied_discount);
+        $this->assertEquals(SummitTicketType::Subtype_Regular, $ticket_types->data[0]->sub_type);
+        return $ticket_types;
+    }
+
     public function testAddTicketType(){
```

Comment:
please provide a test for promo code filter not provided check
https://github.com/OpenStackweb/summit-api/pull/221/files#r1432916790

Reply:
that would be https://github.com/OpenStackweb/summit-api/blob/035df4f50c011e819074777d405be25445ce0f9d/tests/OAuth2SummitTicketTypesApiTest.php#L135